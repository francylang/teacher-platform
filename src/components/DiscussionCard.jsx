import React, { Component } from 'react';
import Comment from './Comment';
import CommentForm from './CommentForm';

class DiscussionCard extends Component {
  constructor () {
    super();
    this.state = {
      displayingComment: false,
    };
  }

  toggleComments() {
    this.setState({
      displayingComment: !this.state.displayingComment,
    });
  }

  showComments() {
    if (this.state.displayingComment) {
      return this.props.comments.map(comment => {
        return (
          <Comment
            key={comment.id}
            comment={comment.comment}
            commentId={comment.discussionId}
            discussionId={this.props.discussion.id}
          />
        )
      })
    }
  }

  showMoreLess() {
    if (this.state.displayingComment) {
      return (
        <p className="show-more-less-text">Hide Comments</p>
      )
    } else {
      return (
        <p className="show-more-less-text">Show Comments</p>
      )
    }
  }

  render() {
    const { title, body } = this.props.discussion
    return (
      <article className="card">
        <section className="top-card">
          <h4 className="card-title">{title}</h4>
          <p className="card-body">{body}</p>
        </section>
        <button
          onClick={() => this.toggleComments()}
          className="show-more-btn">
          {this.showMoreLess()}
        </button>
        {this.showComments()}
      </article>
    );
  }
}

export default DiscussionCard;
