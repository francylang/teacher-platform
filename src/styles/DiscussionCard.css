.card {
  border-radius: 5px;
  background-color: #fff;
  padding: 1% 3% 2% 3%;
  margin-bottom: 2.5%; }

.card-title-container {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center; }

.card-title {
  text-align: left;
  margin: 0;
  display: inline-block;
  font-size: 115%; }

.card-tag {
  border-radius: 5px;
  display: inline-block;
  padding: 4px 6px;
  color: #fff;
  font-size: 90%; }

.R {
  background-color: #FA76C8; }

.E {
  background-color: #7686FA; }

.N {
  background-color: #FABA76; }

.F {
  background-color: #FABA76; }

.G {
  background-color: #FAEA76; }

.S {
  background-color: #86FA76; }

.card-body {
  text-align: left;
  margin-top: 0; }

.show-more-btn {
  background-color: #fff;
  border: none;
  outline: none; }

.show-more-btn:hover {
  cursor: pointer; }

.show-more-less-text {
  text-transform: uppercase;
  color: #5D5D5D;
  margin-bottom: 0px; }

.comments-header {
  text-align: left; }
