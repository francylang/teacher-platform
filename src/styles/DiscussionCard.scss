.card {
  border-radius: 5px;
  background-color: #fff;
  padding: 2.5% 2.5% 10% 2.5%;
}

.card-title {
  text-align: left;
  margin: 0;
}

.card-body {
  text-align: left;
}

.show-more-btn {
  padding: 1% 2.5%;
  float: right;
  width: 135px;
}

.show-more-less-text {
  color: #fff;
  font-family: 'Ubuntu Mono';
  font-size: 90%;
  margin: 5% 0;
}
