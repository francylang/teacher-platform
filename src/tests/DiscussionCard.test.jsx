import React from 'react';
import ReactDOM from 'react-dom';
import DiscussionCard from '../components/DiscussionCard';
import { shallow } from 'enzyme';
import testSetup from './testSetup';

describe('<DiscussionCard />', () => {
  let wrapper;
  let mockFn = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<DiscussionCard />);
  });

  it.only('should exist', () => {
    expect(wrapper).toBeDefined();
  });

  it('should have a false default state', () => {
    expect(wrapper.state().displayingComment).toEqual(false);
  });

  it('should render a card', () => {
    const card = wrapper.find('.card');

    expect(card.length).toEqual(1);
  });

  it('should render a top-card section', () => {
    const topCard = wrapper.find('.top-card');

    expect(topCard.length).toEqual(1);
  });

  it('should render a card title', () => {
    const title = wrapper.find('.card-title');

    expect(title.length).toBeDefined();
  });

  it('should render a card body', () => {
    const body = wrapper.find('.card-body');

    expect(body.length).toBeDefined();
  });

  it('should render a button', () => {
    const button = wrapper.find('.show-more-btn');

    expect(button.length).toBeDefined();
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
