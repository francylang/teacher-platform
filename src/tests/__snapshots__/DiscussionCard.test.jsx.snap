// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DiscussionCard /> should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DiscussionCard
    comments={
        Array [
            Object {
              "comments": Array [
                Object {
                  "comment": "Yes but it's important",
                  "discussionId": 1,
                  "id": 1,
                },
              ],
            },
          ]
    }
    discussion={
        Array [
            Object {
              "body": "Didn't kids get this in Grade 5? I'm confused about why it's in the standards.",
              "id": 1,
              "tagTitle": "6.RP.A.1",
              "title": "Unit Rate",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <section
          className="top-card"
>
          <div
                    className="card-title-container"
          >
                    <h4
                              className="card-title"
                    />
                    <p
                              className="card-tag"
                    />
          </div>
          <p
                    className="card-body"
          />
</section>,
        <button
          className="show-more-btn"
          onClick={[Function]}
>
          <p
                    className="show-more-less-text"
          >
                    More
          </p>
</button>,
        undefined,
        undefined,
        undefined,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="card-title-container"
>
              <h4
                            className="card-title"
              />
              <p
                            className="card-tag"
              />
</div>,
            <p
              className="card-body"
/>,
          ],
          "className": "top-card",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h4
                  className="card-title"
/>,
                <p
                  className="card-tag"
/>,
              ],
              "className": "card-title-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "card-title",
                },
                "ref": null,
                "rendered": null,
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "card-tag",
                },
                "ref": null,
                "rendered": null,
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "card-body",
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
        ],
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <p
            className="show-more-less-text"
>
            More
</p>,
          "className": "show-more-btn",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "More",
            "className": "show-more-less-text",
          },
          "ref": null,
          "rendered": "More",
          "type": "p",
        },
        "type": "button",
      },
      undefined,
      undefined,
      undefined,
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <section
            className="top-card"
>
            <div
                        className="card-title-container"
            >
                        <h4
                                    className="card-title"
                        />
                        <p
                                    className="card-tag"
                        />
            </div>
            <p
                        className="card-body"
            />
</section>,
          <button
            className="show-more-btn"
            onClick={[Function]}
>
            <p
                        className="show-more-less-text"
            >
                        More
            </p>
</button>,
          undefined,
          undefined,
          undefined,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="card-title-container"
>
                <h4
                                className="card-title"
                />
                <p
                                className="card-tag"
                />
</div>,
              <p
                className="card-body"
/>,
            ],
            "className": "top-card",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h4
                    className="card-title"
/>,
                  <p
                    className="card-tag"
/>,
                ],
                "className": "card-title-container",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                    "className": "card-title",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "h4",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                    "className": "card-tag",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "card-body",
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
          ],
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p
              className="show-more-less-text"
>
              More
</p>,
            "className": "show-more-btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "More",
              "className": "show-more-less-text",
            },
            "ref": null,
            "rendered": "More",
            "type": "p",
          },
          "type": "button",
        },
        undefined,
        undefined,
        undefined,
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
